File: backend/springboot/practice/crud/src/main/java/com/example/crud/entity/Document.java
package com.example.crud.entity;

import com.example.crud.dto.CreateDocumentDto;
import com.example.crud.dto.UpdateDocumentDto;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Document {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;
    private String contents;

    public Document(CreateDocumentDto documentDto) {
        title = documentDto.getTitle();
        contents = documentDto.getContents();
    }

}

File: backend/kotlin/crud/src/main/kotlin/com/study/crud/entity/Document.kt
package com.study.crud.entity

import javax.persistence.*

@Entity
class Document(
    title: String,
    content: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    var title = title
        protected set

    var content = content
        protected set

    fun update(title: String, content: String): Document{
        this.title = title
        this.content = content
        return this
    }
}